variables.tf

```diff
@@ -2,8 +2,150 @@
 # Input Variables
 ########################################################################################################################
 
-#variable "my_variable" {
-#  type        = string
-#  description = "A description of my variable"
-#  default     = "default_value"
-#}
+variable "resource_group_id" {
+  description = "The resource group ID where the watsonx services will be provisioned. Required when creating a new instance."
+  type        = string
+  default     = null
+  validation {
+    condition     = var.existing_watson_studio_instance_crn == null ? length(var.resource_group_id) > 0 : true
+    error_message = "You must specify a value for 'resource_group_id', required to create watson studio instance."
+  }
+
+  validation {
+    condition     = var.existing_machine_learning_instance_crn == null ? length(var.resource_group_id) > 0 : true
+    error_message = "You must specify a value for 'resource_group_id', required to create watson machine learning instance."
+  }
+}
+
+variable "prefix" {
+  description = "The name to be used on all Watson resources as a prefix."
+  type        = string
+
+  validation {
+    condition     = var.prefix != "" && length(var.prefix) <= 25
+    error_message = "You must provide a value for the resource_prefix variable and the resource_prefix length can't exceed 25 characters."
+  }
+}
+
+variable "region" {
+  default     = "us-south"
+  description = "Region where the watsonx resources will be provisioned."
+  type        = string
+
+  validation {
+    condition     = contains(["eu-de", "us-south", "eu-gb", "jp-tok", "au-syd"], var.region)
+    error_message = "You must specify `eu-de`, `eu-gb`, `jp-tok` or `us-south` as the IBM Cloud region."
```

Comment:
`au-syd` is missing in the error message.

Reply:
removed au-syd, it is not supported by dataplatform.